packages/bcp/src/cosmwasmconnection.ts

```diff
@@ -130,9 +130,8 @@ export class CosmWasmConnection implements BlockchainConnection {
   }
 
   public async height(): Promise<number> {
-    // tslint:disable-next-line: deprecation
-    const { block } = await this.restClient.blocksLatest();
-    return block.header.height;
+    const { block } = await this.cosmWasmClient.getBlock();
```

Comment:
Note that `getBlock` can be a lot heavier than `getHeader`.

Or at least in some theoretical world, when blocks have many transactions (or if they have a code upload tx)

I think these don't include the signatures (which are HUGE on cosmos hub.... something like 10kB per block), but that is only from the `/commit` endpoint.

In any case, I prefer to getHeader (a small, constant-size return value) when we only need info from the header. And only `getBlock` when we actually want the transactions.

Reply:
I was thinking about this too but our RestClient does not have a function to get headers. And it wurns out, the API does not have an enpoint for headers: https://cosmos.network/rpc/#/

Reply:
Oh man this RestServer misses a lot from raw tendermint..

Good find there, this seems like the best solution, until the refactor everything with protobuf and we can connect directly to tendermint

Reply:
API endpoint [GET /validatorsets/latest (Get the latest validator set)](https://cosmos.network/rpc/#/Tendermint_RPC/get_validatorsets_latest) would be an alternative to get the hight. It might be smaller, depending in the number of validators and transactions in blocks.